modules/core/04-channel/types/msgs.go

```diff
@@ -508,21 +505,17 @@ func (msg MsgChannelUpgradeInit) ValidateBasic() error {
 	if !IsValidChannelID(msg.ChannelId) {
 		return ErrInvalidChannelIdentifier
 	}
-	if msg.ProposedUpgradeChannel.State != INITUPGRADE {
-		return errorsmod.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", INITUPGRADE, msg.ProposedUpgradeChannel.State)
-	}
-	if strings.TrimSpace(msg.ProposedUpgradeChannel.Version) == "" {
-		return errorsmod.Wrap(ErrInvalidChannelVersion, "channel version must not be empty")
-	}
-	if msg.TimeoutHeight.IsZero() && msg.TimeoutTimestamp == 0 {
-		return errorsmod.Wrap(ErrInvalidUpgradeTimeout, "timeout height and timeout timestamp cannot both be 0")
-	}
+
 	_, err := sdk.AccAddressFromBech32(msg.Signer)
 	if err != nil {
 		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
 	}
 
-	return nil
+	if !msg.Timeout.IsValid() {
+		return errorsmod.Wrap(ErrInvalidUpgrade, "upgrade timeout cannot be empty")
```

Comment:
An error message like this would make it similar to [the one for `Packet`](https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/types/packet.go#L107).

```suggestion
		return errorsmod.Wrap(ErrInvalidUpgrade, "upgrade timeout height and upgrade timeout timestamp cannot both be 0")
```